modules/shared-utils/src/cljdoc/util/deps.clj

```diff
@@ -47,15 +47,17 @@
   file but this situation being an edge case this is a sufficient fix for now."
   [pom]
   (->> (pom/dependencies (pom/parse (slurp pom)))
-       (keep (fn [{:keys [group-id artifact-id version]}]
+       (keep (fn [{:keys [group-id artifact-id version scope]}]
                (when-not (or (.startsWith artifact-id "boot-")
                              ;; Ensure that tools.reader version is used as specified by CLJS
                              (and (= group-id "org.clojure")
                                   (= artifact-id "tools.reader"))
                              ;; The version can be nil when pom's utilize dependencyManagement - this unsurprisingly breaks tools.deps
                              ;; Remains to be seen if this causes any issues
                              ;; http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management
-                             (nil? version))
+                             (nil? version)
+                             ;; compile/runtime scopes will be included by the normal dependency resolution.
+                             (not (#{"provided" "system" "test"} scope)))
```

Comment:
Looks great @IamDrowsy! 👍 I find the keep condition `when-not` `or` `not` slightly confusing, maybe it would make sense to refactor those into separate `filter` calls for readability?

Reply:
Just the new line? e.g. adding something like `(filter #(#{"provided" "system" "test"} (:scope %)))` before the `keep` or do you want the refactor the whole `when-not`?

Reply:
Whatever you think is best, we can also keep it this way and defer to fix later. 😄